lib/mapObjects/CBank.cpp

```diff
@@ -95,124 +95,127 @@ bool CBank::wasVisited (PlayerColor player) const
-		if (ID == Obj::PYRAMID) // You come upon the pyramid ... pyramid is completely empty.
-		{
-			iw.text << VLC->generaltexth->advobtxt[107];
-			iw.components.push_back (Component (Component::LUCK, 0 , -2, 0));
-			GiveBonus gb;
-			gb.bonus = Bonus(Bonus::ONE_BATTLE,Bonus::LUCK,Bonus::OBJECT,-2,id.getNum(),VLC->generaltexth->arraytxt[70]);
-			gb.id = h->id.getNum();
-			cb->giveHeroBonus(&gb);
-		}
-		else
+		
+		switch (ID)
 		{
-			iw.text << VLC->generaltexth->advobtxt[33];// This was X, now is completely empty
-			iw.text.addReplacement(getObjectName());
-		}
-		cb->showInfoDialog(&iw);
-	}
-}
-
-void CBank::doVisit(const CGHeroInstance * hero) const
-{
-	int textID = -1;
-	InfoWindow iw;
-	iw.player = hero->getOwner();
-	MetaString loot;
-
-	switch (ID)
-	{
-	case Obj::CREATURE_BANK:
-	case Obj::DRAGON_UTOPIA:
-		textID = 34;
-		break;
-	case Obj::DERELICT_SHIP:
-		if (!bc)
-			textID = 43;
-		else
+		case Obj::SHIPWRECK:
+		case Obj::DERELICT_SHIP:
+		case Obj::CRYPT:
 		{
 			GiveBonus gbonus;
 			gbonus.id = hero->id.getNum();
 			gbonus.bonus.duration = Bonus::ONE_BATTLE;
 			gbonus.bonus.source = Bonus::OBJECT;
 			gbonus.bonus.sid = ID;
-			gbonus.bdescr << "\n" << VLC->generaltexth->arraytxt[101];
 			gbonus.bonus.type = Bonus::MORALE;
 			gbonus.bonus.val = -1;
+			switch (ID)
+			{
+			case Obj::SHIPWRECK:
+				textID = 123;
+				gbonus.bdescr << VLC->generaltexth->arraytxt[99];
+				break;
+			case Obj::DERELICT_SHIP:
+				textID = 42;
+				gbonus.bdescr << VLC->generaltexth->arraytxt[101];
+				break;
+			case Obj::CRYPT:
+				textID = 120;
+				gbonus.bdescr << VLC->generaltexth->arraytxt[98];
+				break;
+			}
 			cb->giveHeroBonus(&gbonus);
-			textID = 42;
-			iw.components.push_back (Component (Component::MORALE, 0 , -1, 0));
+			iw.components.push_back(Component(Component::MORALE, 0, -1, 0));
+			iw.soundID = soundBase::GRAVEYARD;
+			break;
 		}
-		break;
-	case Obj::CRYPT:
-		if (bc)
-			textID = 121;
-		else
+		case Obj::PYRAMID:
 		{
-			iw.components.push_back (Component (Component::MORALE, 0 , -1, 0));
-			GiveBonus gbonus;
-			gbonus.id = hero->id.getNum();
-			gbonus.bonus.duration = Bonus::ONE_BATTLE;
-			gbonus.bonus.source = Bonus::OBJECT;
-			gbonus.bonus.sid = ID;
-			gbonus.bdescr << "\n" << VLC->generaltexth->arraytxt[ID];
-			gbonus.bonus.type = Bonus::MORALE;
-			gbonus.bonus.val = -1;
-			cb->giveHeroBonus(&gbonus);
-			textID = 120;
-			iw.components.push_back (Component (Component::MORALE, 0 , -1, 0));
+			GiveBonus gb;
+			gb.bonus = Bonus(Bonus::ONE_BATTLE, Bonus::LUCK, Bonus::OBJECT, -2, id.getNum(), VLC->generaltexth->arraytxt[70]);
+			gb.id = hero->id.getNum();
+			cb->giveHeroBonus(&gb);
+			textID = 107;
+			iw.components.push_back(Component(Component::LUCK, 0, -2, 0));
+			break;
 		}
-		break;
-	case Obj::SHIPWRECK:
-		if (bc)
-			textID = 124;
-		else
-			textID = 123;
-		break;
-	case Obj::PYRAMID:
-		textID = 106;
+		case Obj::CREATURE_BANK:
+		case Obj::DRAGON_UTOPIA:
+		default:
```

Comment:
What is the point of such case label? Should it better to be just `default`?

Reply:
Just readability, searching for creature bank, or utopia code will be easier with such case.
Not sure if default is needed, can it occur?

Reply:
(Modded) Creature bank can have any ID, it defined by `handler` field.

Reply:
So shouldn't `default` be added to two other switch statements, with some generic text? lines 102 and 142

Reply:
I think it should

Reply:
fixed braces, and` default`